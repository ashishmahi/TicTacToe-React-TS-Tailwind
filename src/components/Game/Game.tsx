import React from "react";

// import { getWinningCombinationFor } from "./gridHelper";

interface GameProps {
  numberOfColumns?: number;
}

const Game: React.FC<GameProps> = ({ numberOfColumns: columnConfig }) => {
  return <div></div>;
};

Game.defaultProps = {
  numberOfColumns: 3,
};

export default Game;
