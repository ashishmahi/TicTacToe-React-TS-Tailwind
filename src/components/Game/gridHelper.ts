
const getHorizontalCombination = (
    griddy: string[][],
    rowIndex: number,
    columnIndex: number
): string[] => {
    const output: string[] = [];
    if (columnIndex == 0) {
        for (let index = 0; index < griddy.length; index++) {
            output.push(griddy[index][rowIndex]);
        }
        return output;
    }
    return output;
}
const getVerticalCombination = (
    griddy: string[][],
    rowIndex: number,
    columnIndex: number
): string[] => {
    const output: string[] = [];
    if (rowIndex == 0) {
        for (let index = 0; index < griddy.length; index++) {
            output.push(griddy[index][columnIndex]);
        }
        return output;
    }
    return output;
}

const getDiagonalCombination = (
    griddy: string[][],
    rowIndex: number,
    columnIndex: number
): string[] => {
    const output: string[] = [];
    const topLeft = rowIndex == 0 && columnIndex == 0;
    const topRight = columnIndex == griddy.length - 1 && rowIndex == 0;
    if (topLeft) {
        for (let index = 0; index < griddy.length; index++) {
            output.push(griddy[rowIndex][columnIndex]);
            rowIndex += 1;
            columnIndex += 1;
        }
        return output;
    }
    if (topRight) {
        for (let index = 0; index < griddy.length; index++) {
            output.push(griddy[rowIndex][columnIndex]);
            rowIndex += 1;
            columnIndex -= 1;
        }
        return output;
    }
    return output;
};

export const getGridFor = (numberOfColumns: number): string[][] => {
    const grid: string[][] = [];
    let row: string[] = [];
    for (let rowIndex = 1; rowIndex <= numberOfColumns; rowIndex++) {
        for (let columnIndex = 1; columnIndex <= numberOfColumns; columnIndex++) {
            row.push(`${rowIndex}_${columnIndex}`);
        }
        grid.push(row);
        row = [];
    }
    return grid;
}
export const getWinningCombinationFor = (numberOfColumns: number): string[][] => {
    const grid = getGridFor(numberOfColumns);
    const result: string[][] = [];
    for (let rowIndex = 0; rowIndex < grid.length; rowIndex++) {
        const row = grid[rowIndex];
        for (let columnIndex = 0; columnIndex < row.length; columnIndex++) {
            const cell = row[columnIndex];
            const horizontalPossibleCombination = getHorizontalCombination(
                grid,
                rowIndex,
                columnIndex
            );
            const diagonalPossibleCombination = getDiagonalCombination(
                grid,
                rowIndex,
                columnIndex
            );
            const verticalPossibleCombination = getVerticalCombination(
                grid,
                rowIndex,
                columnIndex
            );
            if (horizontalPossibleCombination.length) {
                result.push(horizontalPossibleCombination);
            }
            if (verticalPossibleCombination.length) {
                result.push(verticalPossibleCombination);
            }
            if (diagonalPossibleCombination.length) {
                result.push(diagonalPossibleCombination);
            }
        }
    }
    return result;
};
