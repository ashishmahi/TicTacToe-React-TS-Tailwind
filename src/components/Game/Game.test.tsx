import React from "react";
import { render } from "@testing-library/react";
import Game from "./Game";
describe("Game Test", () => {
  it("should render game", () => {
    const { container } = render(<Game />);
    expect(container).toMatchSnapshot();
  });
});
