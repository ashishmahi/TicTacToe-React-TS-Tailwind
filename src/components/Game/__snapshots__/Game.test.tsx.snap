// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Game Test should render game 1`] = `
<div>
  <div />
</div>
`;
